libtcmu_config.c

```diff
@@ -123,12 +123,18 @@ do { \
 	} \
 } while (0)
 
-#define TCMU_PARSE_CFG_STR(cfg, key) \
+#define TCMU_PARSE_CFG_STR(cfg, key, def) \
 do { \
 	struct tcmu_conf_option *option; \
+	char buf[1024]; \
 	option = tcmu_get_option(#key); \
 	if (option) { \
-		cfg->key = strdup(option->opt_str); } \
+		cfg->key = strdup(option->opt_str); \
+		if (option->opt_str) \
```

Comment:
Do you remember what TCMU_FREE_CFG_STR_KEY was for?

Reply:
Yeah, there is a memory leak bug introduced ahead of this PR. I will fix it together with this.
Thanks,